### PR TITLE
Use QgsTextRender to power up copyright decoration styling

### PR DESCRIPTION
**Pull request by nirvn**
## Description
Without further due, a before vs. PR screenshot:
screenshot from 2018-03-27 10-18-54

@nyalldawson , review appreciated.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/qgsdecorationcopyright.cpp [hunk 1]
```diff
@@ -91,77 +107,78 @@ void QgsDecorationCopyright::run()
...
+      yOffset = mMarginVertical;
+      break;
     }
+    case QgsUnitTypes::RenderPercentage:
+    {
+      xOffset = ( ( deviceWidth - textWidth ) / 100. ) * mMarginHorizontal;
+      yOffset = ( ( deviceHeight - textHeight ) / 100. ) * mMarginVertical;
+      break;
+    }
+    default:  // Use default of top left
```
Comment (nyalldawson) [inline src/app/qgsdecorationcopyright.cpp]: Can you remove the default and swap to explicit cases for the remaining values please?

File: src/app/qgsdecorationcopyrightdialog.cpp [hunk 2]
```diff
@@ -70,16 +70,8 @@ QgsDecorationCopyrightDialog::QgsDecorationCopyrightDialog( QgsDecorationCopyrig
...
-  pbnColorChooser->setColor( mDeco.mColor );
-  pbnColorChooser->setContext( QStringLiteral( "gui" ) );
-  pbnColorChooser->setColorDialogTitle( tr( "Select Text color" ) );
-
-  QTextCursor cursor = txtCopyrightText->textCursor();
-  txtCopyrightText->selectAll();
-  txtCopyrightText->setTextColor( mDeco.mColor );
-  txtCopyrightText->setTextCursor( cursor );
+  // font settings
+  mButtonFontStyle->setTextFormat( mDeco.textFormat() );
```
Comment (nyalldawson) [inline src/app/qgsdecorationcopyrightdialog.cpp]: Can you also:

- set the font button dialog title
- set the font button map canvas (i figure why not... it'll unlock some extra canvas related functionality in the dialog, which may not be terribly useful here, but at the cost of a single line of code may come in useful for 1 or 2 users!)

File: src/core/qgstextrenderer.cpp [hunk 3]
```diff
@@ -2005,7 +2005,7 @@ void QgsTextRenderer::drawBackground( QgsRenderContext &context, QgsTextRenderer
 
       case Point:
       {
-        double originAdjust = fm.ascent() / 2.0 - fm.leading() / 2.0;
+        double originAdjust = fm.ascent() / 2.0 - fm.leading();
```
Comment (nyalldawson) [inline src/core/qgstextrenderer.cpp]: What's the rationale here?

**Review by nyalldawson — COMMENTED** [review]
Nice - great use of text renderer!

Two questions:

- we lose html support here. I don't see that as a big issue personally, but maybe a check box is needed to allow it?
- should we restore existing color setting? I noticed it's discarded here

**Comment by nirvn** [comment]
@nyalldawson , thanks for the quick review, comments addressed.

As for HTML support, I'd rather have HTML support back through QgsTextRenderer getting HTML support instead. @NathanW2 was working on this a while back.

**Comment by nirvn** [comment]
I'll merge that this afternoon if nobody objects.

**Pull request merged**